core/appmodule/module.go

```diff
@@ -38,6 +38,19 @@ type HasServices interface {
 	RegisterServices(grpc.ServiceRegistrar) error
 }
 
+// PrepareCheckStateAppModule is an extension interface that contains information about the AppModule
+// and PrepareCheckState.
+type PrepareCheckStateAppModule interface {
```

Comment:
While you are at it, could you rename them to `HasPrepareCheckState` and `HasPrecommit ` respectively?
This makes more sense with the other interface here and given the import path there is no `AppModule` repetition. If you are already doing that, then nvm 🙈

Reply:
yea works, will make the change